### ISSUE TITLE
Bug:  Electron app crash (NotFoundError Failed to execute 'removeChild' on 'Node')

### ISSUE THREAD

**Issue by Tuddual** [facebook/react#21772]
Hello, I am coding an app with electron and react (based on the [electron-react-boilerplate](https://github.com/electron-react-boilerplate/electron-react-boilerplate)), my code isn't perfect but it runs almost as I want.
The last version on my code is here: https://github.com/Tuddual/word2html/tree/v1.1.2

React version: 17.0.1
Electron version: 11.0.1

## Steps To Reproduce

The problem append with I trigger [this function](https://github.com/Tuddual/word2html/blob/ed382dc8adf103e63dd349ffa4d79220096a6ade/src/components/App.tsx#L91) :
```
const saveHTML = () => {
    const html = output(info.content, info.style);
    const filePath = path.join(info.path, '/my_report.html');
    setPath(filePath);

    fs.writeFile(filePath, html, (err: any) => {
      if (err) setOpenAlert1(true);
      else setOpenAlert2(true);
    });
  };
```

The problem is that my function works perfectly as wanted **but** when I previously remove a child element with [a function](https://github.com/Tuddual/word2html/blob/ed382dc8adf103e63dd349ffa4d79220096a6ade/src/components/Custom/editBox.tsx#L277)
```
const deleteBox = () => {
    ...
    // remove the box
    const child = document.getElementById(id)!.parentElement;
    child!.parentElement!.removeChild(child!);
  };
```
then when I trigger the function `saveHTML` this error append :
```
Uncaught DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
    at removeChild (http://localhost:1212/dist/renderer.dev.js:65859:18)
    at unmountHostComponents (http://localhost:1212/dist/renderer.dev.js:76849:9)
    at commitDeletion (http://localhost:1212/dist/renderer.dev.js:76900:5)
    at commitMutationEffects (http://localhost:1212/dist/renderer.dev.js:78960:11)
    at HTMLUnknownElement.callCallback (http://localhost:1212/dist/renderer.dev.js:59503:14)
    at Object.invokeGuardedCallbackDev (http://localhost:1212/dist/renderer.dev.js:59552:16)
    at invokeGuardedCallback (http://localhost:1212/dist/renderer.dev.js:59614:31)
    at commitRootImpl (http://localhost:1212/dist/renderer.dev.js:78674:9)
    at unstable_runWithPriority (http://localhost:1212/dist/renderer.dev.js:88221:12)
    at runWithPriority$1 (http://localhost:1212/dist/renderer.dev.js:66834:10)
```
```
The above error occurred in the <div> component:

    at div
    at div
    at div
    at StepThree (http://localhost:1212/dist/main.382204ce087752fd6518.hot-update.js:1304:17)
    at ThemeProvider (http://localhost:1212/dist/renderer.dev.js:19610:24)
    at div
    at App (http://localhost:1212/dist/main.093915cc12adb5dbc6e8.hot-update.js:116:17)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
```

I don't know what am I missing and don't know if this is the right place to ask for help, but thanks for reading.

**Comment by bvaughn** [comment]
I'm not sure what's going on here, but it seems most likely that your `deleteBox` function is modifying DOM that was originally rendered by React? React doesn't support this. DOM rendered by React should only be updated by React. (If you want to remove some of the DOM, you can schedule an update with React that conditionally renders the element you want to remove.)

**Comment by Tuddual** [comment]
Hello @bvaughn thanks for answering me,

The problems seem to come from the [Alert component](https://material-ui.com/components/alert/) from material-ui that I used. It is just a supposition but when I didn't use the Alert component in my app there was no error.

The function `saveHTML` was using `setOpenAlert1(true)` or `setOpenAlert2(true)` and this was opening the alert :
```
<Snackbar
    anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
    open={openAlert1}                                         // <-- setOpenAlert1(true) going to set openAlert1 on true
    autoHideDuration={2000}
    onClose={handleClose1}
>
    <Alert onClose={handleClose1} severity="error">
        The HTML file failed to be saved here : {pathHTML}
    </Alert>
</Snackbar>
<Snackbar
    anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
    open={openAlert2}                                        // <-- setOpenAlert2(true) going to set openAlert2 on true
    autoHideDuration={2000}
    onClose={handleClose2}
>
    <Alert onClose={handleClose2} severity="success">
        The HTML file has been successfully saved here: {pathHTML}
    </Alert>
</Snackbar>
```

I understood what you are saying (about not modifying DOM that was originally rendered by React).

What I am going to plan to sort this out :
 - update my code to not modifying DOM that was originally rendered by React
 - and if the error persists after my update, I won't use the Alert component.

**Comment by Tuddual** [comment]
Hello,

I did try to not modifying the DOM that was originally rendered by React but when I did that the rerender was taking too long time (1sec). The method I used wasn't optimal but I didn't find any other solution and my project is a time-limited project, therefore, I need to move on fast.

[My solution](https://github.com/Tuddual/word2html/pull/83) still modifying the DOM (sorry for the nasty solution) but as I was saying 2 days ago, the solution I used is to not use the [snackbars](https://material-ui.com/components/snackbars/) and [alert](https://material-ui.com/components/alert/) component from material UI.

Now I am not using those components, to alert my user I just add a green check/ red cross, it's less beautiful but it works fine from my perspective.
